src/librustc_session/options.rs

```diff
@@ -0,0 +1,446 @@
+use crate::config::*;
+
+use crate::lint;
+use crate::utils::NativeLibraryKind;
+use crate::early_error;
+use crate::search_paths::SearchPath;
+
+use rustc_target::spec::{LinkerFlavor, MergeFunctions, PanicStrategy, RelroLevel};
+use rustc_target::spec::TargetTriple;
+
+use syntax_pos::edition::Edition;
+use rustc_feature::UnstableFeatures;
+
+use getopts;
+
+use std::collections::BTreeMap;
+
+use std::str;
+use std::hash::Hasher;
+use std::collections::hash_map::DefaultHasher;
+use std::path::PathBuf;
+
+// The top-level command-line options struct.
+//
+// For each option, one has to specify how it behaves with regard to the
+// dependency tracking system of incremental compilation. This is done via the
+// square-bracketed directive after the field type. The options are:
+//
+// [TRACKED]
+// A change in the given field will cause the compiler to completely clear the
+// incremental compilation cache before proceeding.
+//
+// [UNTRACKED]
+// Incremental compilation is not influenced by this option.
+//
+// If you add a new option to this struct or one of the sub-structs like
+// `CodegenOptions`, think about how it influences incremental compilation. If in
+// doubt, specify [TRACKED], which is always "correct" but might lead to
+// unnecessary re-compilation.
+top_level_options!(
```

Comment:
Let's move the macros as well (this one and `options!`)?

Reply:
Sure, updated.